app/src/full/java/io/homeassistant/companion/android/sensors/CarInfoSensorManager.kt

```diff
@@ -0,0 +1,116 @@
+package io.homeassistant.companion.android.sensors
+
+import android.content.Context
+import android.os.Build
+import android.util.Log
+import androidx.annotation.RequiresApi
+import androidx.car.app.hardware.common.CarValue
+import androidx.car.app.hardware.common.OnCarDataAvailableListener
+import androidx.car.app.hardware.info.EnergyLevel
+import io.homeassistant.companion.android.common.R
+import io.homeassistant.companion.android.common.sensors.SensorManager
+import io.homeassistant.companion.android.vehicle.HaCarAppService
+
+class CarInfoSensorManager: SensorManager, OnCarDataAvailableListener<EnergyLevel> {
+
+    companion object {
+        private const val TAG = "CarInfoSM"
+        private var isListenerRegistered = false
+        private var listenerLastRegistered = 0
+        private val fuelLevel = SensorManager.BasicSensor(
+            "car_info_fuel_level",
+            "sensor",
+            R.string.basic_sensor_name_car_info_fuel_level,
+            R.string.sensor_description_car_info_fuel_level,
+            "mdi:barrel",
+            unitOfMeasurement = "%",
+            stateClass = SensorManager.STATE_CLASS_MEASUREMENT,
+        )
+        private val batteryLevel = SensorManager.BasicSensor(
+            "car_info_battery_level",
+            "sensor",
+            R.string.basic_sensor_name_car_info_battery_level,
+            R.string.sensor_description_car_info_battery_level,
+            "mdi:car-battery",
+            unitOfMeasurement = "%",
+            stateClass = SensorManager.STATE_CLASS_MEASUREMENT,
+        )
+    }
+    private lateinit var latestContext: Context
+
+    override fun docsLink(): String {
+        return "https://companion.home-assistant.io/docs/core/sensors"
+    }
+
+    override fun hasSensor(context: Context): Boolean {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.P
+    }
+
+    override val name: Int
+        get() = R.string.sensor_name_car_info
+
+    override suspend fun getAvailableSensors(context: Context): List<SensorManager.BasicSensor> {
+        return listOf(
+            fuelLevel, batteryLevel
+        )
+    }
+
+    override fun requiredPermissions(sensorId: String): Array<String> {
+        return arrayOf("com.google.android.gms.permission.CAR_FUEL")
```

Comment:
Is this also required for the battery sensor or just the fuel sensor?

Reply:
Yes, both are provided by the same callback, [here](https://developer.android.com/training/cars/apps#carinfo) is the list of required permissions. We may want to filter on the sensorId if we add other listeners like mileage that requires different permissions.